src/submodules/io/jld2_integration.jl

```diff
@@ -0,0 +1,199 @@
+    )
+
+function from_serializable(t::SerializableABM{S,A}; kwargs...) where {S,A}
+    abm = ABM(
+        A,
+        from_serializable(t.space; kwargs...);
+        scheduler = get(kwargs, :scheduler, Schedulers.fastest),
+        properties = from_serializable(t.properties; kwargs...),
+        rng = t.rng,
+    )
+    abm.maxid[] = t.maxid
+    for a in t.agents
+        add_agent_pos!(a, abm)
+    end
+    return abm
+end
+
+function from_serializable(t::SerializableGridSpace{D,P,W}; kwargs...) where {D,P,W}
+    s = Array{Vector{Int},D}(undef, t.dims)
+    for i in eachindex(s)
+        s[i] = Int[]
+    end
+    pathfinder = from_serializable(t.pathfinder; kwargs...)
+    return GridSpace{D,P,typeof(pathfinder)}(
+        s,
+        t.metric,
+        Dict(k => v for (k, v) in t.hoods),
+        Dict(k => v for (k, v) in t.hoods_tuple),
+        pathfinder,
+    )
+end
+
+function from_serializable(t::SerializableContinuousSpace{D,P,T}; kwargs...) where {D,P,T}
+    update_vel! = get(kwargs, :update_vel!, Agents.defvel)
+    ContinuousSpace(
+        from_serializable(t.grid; kwargs...),
+        update_vel!,
+        t.dims,
+        t.spacing,
+        t.extent,
+    )
+end
+
+from_serializable(t::SerializableGraphSpace; kwargs...) = GraphSpace(t.graph)
+
+from_serializable(t::SerializableAStar{D,P,M}; kwargs...) where {D,P,M} =
+    Pathfinding.AStar{D,P,M}(
+        Dict{Int,Pathfinding.Path{D}}(
+            k => Pathfinding.Path{D}(v...) for (k, v) in t.agent_paths
+        ),
+        t.grid_dims,
+        map(CartesianIndex, t.neighborhood),
+        t.admissibility,
+        t.walkable,
+        t.cost_metric,
+    )
+
+"""
+    AgentsIO.save_checkpoint(filename, model::ABM)
+
+Write the entire `model` to the JLD2 file specified by `filename`. Agent data, including
+multi-agent models, is also saved. Serialization capability depends on JLD2.jl.
+Currently, serialization is also not supported for models using OpenStreetMapSpace.
+Functions are not saved, including stepping functions, schedulers, and `update_vel!`.
+The last two can be provided to [`AgentsIO.load_checkpoint`](@ref) using the appropriate
+keyword arguments. In case you require custom serialization for model properties,
+refer to [`AgentsIO.to_serializable`](@ref) and [`AgentsIO.from_serializable`](@ref).
+"""
+function save_checkpoint(filename, model::ABM)
+    @assert !(model.space isa OpenStreetMapSpace) "Currently serialization is not supported for OpenStreetMapSpace"
+    model = to_serializable(model)
+    @save filename model
+end
+
+"""
+    AgentsIO.load_checkpoint(filename; kwargs...)
+
+Load the model saved to the file specified by `filename`.
+
+The keyword argument `scheduler = Schedulers.fastest` specifies what scheduler should
+be used for the model.
+
+The keyword argument `update_vel!` specifies a function that should be used to
+update each agent's velocity before it is moved. Refer to [`ContinuousSpace`](@ref) for
+details.
+
```

Comment:
Think we should add `warn` to the handled `kwargs`. It's best to keep `warn = true` as default to make sure the loading passes checks, but if I know I have a multi-agent model, I'd prefer to suppress the warning message on-load.